### PR TITLE
Poll in background for changes the set of tables and trigger reconfiguration when tables are added or removed.

### LINKED ISSUE #5: Support dynamic sets of tables
If the user does a CREATE TABLE or DROP TABLE, the Connector should pick this up and reconfigure tasks appropriately. This requires some polling in the Connector on a background thread to periodically check whether the table set has changed.

As part of this, we should verify the Tasks gracefully handle the DROP TABLE case. This should just result in a SQLException which we need to handle anyway, but we might be able to handle this specific case more gracefully if we can narrow the exception to being caused by a dropped table.

### PR DESCRIPTION
**Pull request by ewencp**
Fixes #5.

### CODE REVIEW

**Comment by ewencp** [comment]
@Ishiihara Care to review? This moves all the table monitoring at the connector level to a background thread that can be unit tested separately from the rest of the connector functionality. I left a TODO in there for additional filtering when #3 is implemented (and probably further adjustment will be needed for #2). Unfortunately no system tests for this yet, we'll want to add those to muckrake after we add the baseline tests with a fixed set of tables.

File: src/main/java/io/confluent/copycat/jdbc/JdbcSourceConnector.java [hunk 1]
```diff
@@ -70,28 +75,30 @@ public void start(Properties properties) throws CopycatException {
...
     }
+    return taskConfigs;
   }
 
   @Override
   public void stop() throws CopycatException {
+    log.info("Stopping table monitoring thread");
+    tableMonitorThread.shutdown();
+    try {
+      tableMonitorThread.join(10000);
```
Comment (Ishiihara) [inline src/main/java/io/confluent/copycat/jdbc/JdbcSourceConnector.java]: Shall we define an instance variable for the timeout?

File: src/main/java/io/confluent/copycat/jdbc/TableMonitorThread.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          return;
+        }
+      } catch (InterruptedException e) {
+        log.error("Unexpected InterruptedException, ignoring: ", e);
+      }
+    }
+  }
+
+  public List<String> tables() {
+    final long TIMEOUT = 10000;
```
Comment (Ishiihara) [inline src/main/java/io/confluent/copycat/jdbc/TableMonitorThread.java]: 10000L?

**Comment by ewencp** [comment]
@Ishiihara Anything else to comment on here or is this good to merge?

**Pull request merged**